src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,38 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> eventListeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executorService = Executors.newCachedThreadPool();
 
+    public void registerListener(EventListener listener) {
+        eventListeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        eventListeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : eventListeners) {
+            executorService.submit(() -> listener.onEvent(event));
+        }
     }
 
     public void shutdown() {
-
+        executorService.shutdown();
+        try {
+            if (!executorService.awaitTermination(60, TimeUnit.SECONDS)) {
+                executorService.shutdownNow();
+            }
+        } catch (InterruptedException ie) {
+            executorService.shutdownNow();
```

Comment:
When catching InterruptedException, it's important to preserve the interrupt status by calling Thread.currentThread().interrupt() immediately. This ensures that higher-level interrupt handlers will be aware of the interrupt. The current implementation does call interrupt(), but it's done after calling shutdownNow(), which might be confusing. Consider restructuring the code to call interrupt() as soon as the exception is caught.